Helper/Cart.php

```diff
@@ -2354,9 +2354,11 @@ public function createCart($items, $metadata = null)
         }
 
         $quote->setIsActive(false);
-
+        $this->saveQuote($quote);
         $cart_data = $this->getCartData(false, '', $quote);
         $this->quoteResourceSave($quote);
+        $this->saveQuote($quote);
```

Comment:
Why is it necessary to call `$this->saveQuote($quote);` twice?

Reply:
We first add products to the quote, if we do not call `$this->saveQuote($quote)`; before `$cart_data = $this->getCartData(false, '', $quote);`, then `$quoteRepository->get($quoteId)->getItemsCount` returns 0 which means the items are not saved properly into quoteRepository (\Magento\Quote\Api\CartRepositoryInterface).

Similarly, since the code in the function getCartData may update the quote, we need to call `$this->saveQuote($quote)` again to make sure everything is saved.

Then in my opinion, quoteResourceSave is to talk to the database while saveQuote is for run-time logic, so they are both needed at the end.

Reply:
I am not sure it makes sense, but maybe we can move saveQuote into the exact place where are we updating the quote? Let me know what do you think

Reply:
@vitaliyreznikov What do you mean "the exact place"? every time when the quote is updatd?

Reply:
Ok, I just thought we didn't update the quote each time, but now I see that it's not the case. LGTM.